types/relay-runtime/lib/store/RelayModernEnvironment.d.ts

```diff
@@ -25,43 +24,51 @@ import { RelayObservable } from '../network/RelayObservable';
 export interface EnvironmentConfig {
     readonly configName?: string;
     readonly handlerProvider?: HandlerProvider | null;
+    readonly treatMissingFieldsAsNull?: boolean;
     readonly log?: LogFunction | null;
     readonly operationLoader?: OperationLoader | null;
     readonly network: Network;
     readonly scheduler?: TaskScheduler | null;
     readonly store: Store;
     readonly missingFieldHandlers?: ReadonlyArray<MissingFieldHandler> | null;
     readonly operationTracker?: OperationTracker | null;
+    readonly options?: any;
+    readonly isServer?: boolean;
 }
 
-export class RelayModernEnvironment implements Environment {
+export default class RelayModernEnvironment implements Environment {
+    options: any;
```

Comment:
I wonder if it makes sense to make this generic, so the user can provide a typing for their custom options?

E.g.

```ts
export default class RelayModernEnvironment<Options = undefined> implements Environment {
  options: Options
}
```

Does upstream do anything like that?

Reply:
`mixed`

Reply:
https://flow.org/en/docs/types/mixed/
`Anything goes in, Nothing comes out`

Given I don't know what else the Generic would do, I'd stick up for `unknown`, but if anyone thinks it would be useful, then yeah, generics are awesome

Reply:
I guess a better question to be answered first is, how does this end up being used? I assume it’s so you can refer to that data later on in e.g. your network layer, where you use mutations, etc?

Reply:
This is an excellent question ... I admit that I discovered this configuration with this PR :)

Reply:
> Given I don't know what else the Generic would do, I'd stick up for `unknown`, but if anyone thinks it would be useful, then yeah, generics are awesome

So if it’s metadata meant only for _you_ the author of the project, then I can imagine that it could be useful to do something like:

```ts
interface EnvironmentConfig<Options> {
    options: Options
}

class Environment<Options = undefined> {
    options: Options

    constructor(config: EnvironmentConfig<Options>) {
        this.options = config.options
    }
}

const MyEnvironment = new Environment({ options: { myCustomOptions: 42 } })

// Later on use the environment's options:
MyEnvironment.options.myCustomOptions + 42;
```

https://www.typescriptlang.org/play/index.html?ssl=1&ssc=1&pln=16&pc=44#

Reply:
At the moment I would opt for unknown and then propose a PR to make the options generic  in relay-runtime

Reply:
seconded.  Kick this can down the road, when someone needs it/it becomes more clear

Reply:
Sounds good 👍